api/src/contexts/users/controllers/UpdateUserProfileController.ts

```diff
@@ -0,0 +1,34 @@
+import express from 'express';
+import { IUpdateUserProfileInteractor } from '../usecases/IUpdateUserProfileInteractor';
+import { UpdateUserProfileInput } from '../interactors/UpdateUserProfileInteractor';
+
+export class UpdateUserProfileController {
+  constructor(
+    private readonly updateUserProfileInteractor: IUpdateUserProfileInteractor,
+  ) {}
+
+  async execute(req: express.Request, res: express.Response) {
+    try {
+      const userId = parseInt(req.params.id, 10);
+      if (isNaN(userId)) {
+        return res.status(400).json({ message: 'User ID must be a number' });
+      }
```

Comment:
ここで、DeleteUserControllerにあるように、認証済みのユーザーか確認できるようにできればいいような気がします！

```suggestion
    const authenticatedUser = req.user;
          if (!authenticatedUser) {
            return res.status(401).json({ message: 'Authentication required' });
          }
```